### PR TITLE
Converting Deployment Manager code to TerraForm. Some of DM files hav…

### PR DESCRIPTION
**Pull request by homernajafi**
…e not been deleted because gke-to-gke-peering is still using them.

This Terraform code can be improved by using TF Loops but it's a time consuming task (in case we want to assign time to it)

### CODE REVIEW

File: gke-to-gke-vpn/terraform/main.tf [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+Creating Clusters and their Node Pools
+*/
+
+# Gets the current version of Kubernetes engine
+data "google_container_engine_versions" "gke_version" {
+  location = "us-east1-b"
+  project  = "${var.project}"
+}
+
+resource "google_container_cluster" "cluster-deployment-cluster1" {
```
Comment (chrislovecnm) [inline gke-to-gke-vpn/terraform/main.tf:28]: We need code level documentation on all of the terraform.

File: gke-to-gke-vpn/terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+limitations under the License.
+*/
+
+/*
+Creating Clusters and their Node Pools
+*/
+
+# Gets the current version of Kubernetes engine
+data "google_container_engine_versions" "gke_version" {
+  location = "us-east1-b"
```
Comment (chrislovecnm) [inline gke-to-gke-vpn/terraform/main.tf:23]: Can we make this a variable?

File: gke-to-gke-vpn/terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  project  = "${var.project}"
+}
+
+resource "google_container_cluster" "cluster-deployment-cluster1" {
+  name               = "cluster-deployment-cluster1"
+  project            = "${var.project}"
+  location           = "${var.cluster1-location}"
```
Comment (chrislovecnm) [inline gke-to-gke-vpn/terraform/main.tf:31]: Conceal meta data?  What else are we missing?

File: gke-to-gke-vpn/terraform/network.tf [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+/*
+This will setup basic underlying network vpn.tf
+It will setup Networks, IPs and Subnetworks
+*/
+
+resource "google_compute_address" "vpn1-ip-address" {
+  name    = "vpn1-ip-address"
+  project = "${var.project}"
+  region  = "us-east1"
```
Comment (chrislovecnm) [inline gke-to-gke-vpn/terraform/network.tf]: again lets make this a variable, and set two different regions.

File: gke-to-gke-vpn/terraform/variables.tf [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+/*
+Variables for the creation of the cluster, bastion, subnet and network.
+See https://www.terraform.io/docs/configuration/variables.html
+*/
+
+variable "project" {
+  description = "The project in which to hold the components"
+  type        = "string"
+#  default     = "homer-proj"
```
Comment (chrislovecnm) [inline gke-to-gke-vpn/terraform/variables.tf]: remove

**Review by chrislovecnm — CHANGES_REQUESTED** [review]
Looks good mostly nit picks :)

**Comment by chrislovecnm** [comment]
@homernajafi CI is failing :(

**Comment by homernajafi** [comment]
whitespace issue. good comments. I will fix those.

**Comment by homernajafi** [comment]
test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Comment by homernajafi** [comment]
Test this please

**Review by chrislovecnm — COMMENTED** [review]
Can we add back in deployment manager, and see if we can get that passing, otherwise we will disable for now. 

Also we need to update the documentation for the prerequisites, and talk about terraform.  Please check with Kristen if we need to update a quicklab that corresponds to this as well.

**Comment by homernajafi** [comment]
CI is passing. I will talk to Kristen about doc. I will try to finish before I leave.

**Review by chrislovecnm — APPROVED** [review]

**Comment by chrislovecnm** [comment]
@homernajafi I am going to merge, and let you work on docs in a separate PR.

**Pull request merged**